### PR TITLE
SONAR-9616 propagate visibility to branches

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

**Comment by julienlancelot** [comment]
Commits are missing the JIRA ticket 9616

File: server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 1]
```diff
@@ -354,6 +354,10 @@ public void resetBChangedForRootComponentUuid(DbSession session, String projectU
 
   public void setPrivateForRootComponentUuid(DbSession session, String projectUuid, boolean isPrivate) {
     mapper(session).setPrivateForRootComponentUuid(projectUuid, isPrivate);
+    session.getMapper(BranchMapper.class).selectByProjectUuid(projectUuid)
```
Comment (julienlancelot) [inline server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java]: I thought we said that you'll move it to the WS ?

File: server/sonar-server/src/test/java/org/sonar/server/project/ws/UpdateVisibilityActionTest.java [hunk 2]
```diff
@@ -280,17 +281,24 @@ public void execute_throws_ISE_when_project_organization_uuid_does_not_match_exi
   public void execute_changes_private_flag_of_specified_project_and_all_children_to_specified_new_visibility() {
     ComponentDto project = randomPublicOrPrivateProject();
     boolean initiallyPrivate = project.isPrivate();
+
+    BranchDto branchDto = ComponentTesting.newBranchDto(project);
+    dbClient.branchDao().insert(dbSession, branchDto);
+
+    ComponentDto branch = ComponentTesting.newProjectBranch(project, branchDto);
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/project/ws/UpdateVisibilityActionTest.java]: You should add new test cases for test on branches

**Comment by janos-ss** [comment]
I'll make sure the merge commit will have the JIRA ticket, and then it will be correctly linked from JIRA (like in previous PRs). The commits are ephemeral anyway, as I will squash them when I merge.

File: server/sonar-server/src/test/java/org/sonar/server/project/ws/UpdateVisibilityActionTest.java (hunk 2 above)
Comment (janos-ss) [inline server/sonar-server/src/test/java/org/sonar/server/project/ws/UpdateVisibilityActionTest.java]: Do you mean, to verify that new branches inherit the same visibility from the project? That was already the case before my change, so I assumed that was tested. Or do you mean something else?

**Pull request merged**